ctriface/manual_cleanup_test.go

```diff
@@ -329,5 +380,117 @@ func TestParallelPhasedSnapLoad(t *testing.T) {
 		vmGroup.Wait()
 	}
 
+	{
+		var vmGroup sync.WaitGroup
+		for i := 0; i < vmNum; i++ {
+			vmGroup.Add(1)
+			go func(i int) {
+				defer vmGroup.Done()
+				vmID := fmt.Sprintf("%d", i+vmIDBase)
+				err := orch.StopSingleVM(ctx, vmID)
+				require.NoError(t, err, "Failed to stop VM, "+vmID)
+			}(i)
+		}
+		vmGroup.Wait()
+	}
+
+	orch.Cleanup()
+}
+
+func TestRemoteSnapCreate(t *testing.T) {
+	// Needs to be cleaned up manually.
+	log.SetFormatter(&log.TextFormatter{
+		TimestampFormat: ctrdlog.RFC3339NanoFixed,
+		FullTimestamp:   true,
+	})
+	//log.SetReportCaller(true) // FIXME: make sure it's false unless debugging
+
+	log.SetOutput(os.Stdout)
+
+	log.SetLevel(log.InfoLevel)
+
+	testTimeout := 120 * time.Second
+	ctx, cancel := context.WithTimeout(namespaces.WithNamespace(context.Background(), namespaceName), testTimeout)
+	defer cancel()
+	
+	vmID := "37"
+	revision := "myrev-37"
+
+	err := os.MkdirAll(remoteSnapshotsDir, 0755)
+	require.NoError(t, err, "Failed to create remote snapshots directory")
+
+	orch := NewOrchestrator(
+		"devmapper",
+		"",
+		WithTestModeOn(true),
+		WithUPF(*isUPFEnabled),
+		WithLazyMode(*isLazyMode),
+	)
+
+	_, _, err = orch.StartVM(ctx, vmID, testImageName)
+	require.NoError(t, err, "Failed to start VM")
+
+	err = orch.PauseVM(ctx, vmID)
+	require.NoError(t, err, "Failed to pause VM")
+
+	snap := snapshotting.NewSnapshot(revision, remoteSnapshotsDir, testImageName)
+	_ = snap.Cleanup()
+	err = snap.CreateSnapDir()
+	require.NoError(t, err, "Failed to create remote snapshots directory")
+
+	err = orch.CreateSnapshot(ctx, vmID, snap)
+	require.NoError(t, err, "Failed to create snapshot of VM")
+
+	_, err = orch.ResumeVM(ctx, vmID)
+	require.NoError(t, err, "Failed to resume VM")
+
+	err = orch.StopSingleVM(ctx, vmID)
+	require.NoError(t, err, "Failed to offload VM")
+	
 	orch.Cleanup()
 }
+
+func TestRemoteSnapLoad(t *testing.T) {
```

Comment:
can `TestRemoteSnapLoad` used w/o `TestRemoteSnapCreate`? if not, I suggest replicating the snap creation procedure

Reply:
No, it cannot be. The `TestRemoteSnapCreate` test produces a snapshot artifact which is consequently used by the `TestRemoteSnapLoad` test. 

See also my comment in DM.